### ISSUE TITLE
model.layers.pop() does not work properly on mac

### ISSUE THREAD

**Issue by sfilatye** [fchollet/keras#3199]
I am replacing the last softmax layer (with 1000 output units) of the original model (VGG16) with a new softmax layer (with 10 output units):
    model.add(Dense(4096, activation='relu'))
    model.add(Dropout(0.5))
    model.add(Dense(1000, activation='softmax'))
    model.load_weights('./vgg16_weights.h5')
    model.layers.pop()
    model.layers.pop()
    model.add(Dropout(0.5))
    model.add(Dense(10, activation='softmax'))
    sgd = SGD(lr=1e-3, decay=1e-6, momentum=0.9, nesterov=True)
    model.compile(optimizer=sgd, loss='categorical_crossentropy')

However, when I check input_shape and output_shape of the last three layers I find:

> > > model.layers[-1].output_shape
> > > (None, 10)
> > > model.layers[-1].input_shape
> > > (None, 1000)
> > > model.layers[-2].input_shape
> > > (None, 1000)
> > > model.layers[-2].output_shape
> > > (None, 1000)
> > > model.layers[-3].output_shape
> > > (None, 4096)
> > > model.layers[-2].name
> > > 'dropout_3'
> > > model.layers[-3].name
> > > 'dense_2'
> > > model.layers[-1].name
> > > 'dense_4'

Apparently, model.layers.pop() does not fully update the model. I have OS X 10.11.5 and I just updated  keras and tensorflow (I cannot use Theano since it produces  Illegal instruction: 4 error)

**Comment by Sekunde** [comment]
I have the same problem.
Logically, the following 2 parts of code should be the same in context of pre-trained VGG-16(ending part).
Part 1:
model.add(Dense(1000, activation='softmax'))
model.layers.pop()
model.add(Dense(10, activation='softmax'))
Part 2:
model.add(Dense(10, activation='softmax'))

But part 1 achieves 20% accuracy while part2 achieves 98% over the same training and test datasets.

**Comment by ChristianThomae** [comment]
Duplicate of #2371 - `layers.pop()` is a hack and does not do what you think it does.

**Comment by farizrahman4u** [comment]
#3252
